src/components/UI/SearchInput.js

```diff
@@ -0,0 +1,53 @@
+import styles from "./SearchInput.module.css";
+import { useEffect, useState } from "react";
+import { useRef } from "react";
+
+const DEFAULT_DELAY = 1000;
+
+const SearchInput = ({ onChange, delay }) => {
+  const [value, setValue] = useState("");
+  const [isEmpty, setIsEmpty] = useState(true);
+  const timeoutId = useRef();
```

Comment:
How many search inputs do you plan to have? DO you need separate debounce for each of them? 

If not, why don't use shared `let searchInputTimeout` outside the component? :)